supervised_learning/requirements.txt

```diff
@@ -0,0 +1,22 @@
+# requirements.txt
+
+numpy==1.23.5
```

Comment:
Is there any reason to pin this version to 1.23.5? I would either suggest setting a minimum version for every package or pinning everything. If you mix both, the dependencies will break in the future.